src/python/test/test_extract_expression.py

```diff
@@ -236,6 +268,146 @@ def common_input_args_test(self, input_argument_dict, expected_error_message):
+        with self.assertRaises(ValueError) as cm:
+            write_expression_output(
+                arg_output="-",
+                arg_delim=",",
+                arg_sql=True,
+                output_listdict_or_string=["not a string-formatted SQL query"],
+                error_handler=self.common_value_error_handler
+            )
+        err_msg = str(cm.exception).strip()
+        self.assertEqual(expected_error_message, err_msg)
+
+    def test_output_bad_delimiter(self):
+        bad_delim = "|"
+        expected_error_message =  "Unsupported delimiter: {}".format(bad_delim)
+        with self.assertRaises(ValueError) as cm:
+            write_expression_output(
+                arg_output= "-",
+                arg_delim=bad_delim,
+                arg_sql=False,
+                output_listdict_or_string=self.vizserver_data_mock_response["results"],
+                save_uncommon_delim_to_txt = False,
+                error_handler=self.common_value_error_handler
+            )
+        err_msg = str(cm.exception).strip()
+        self.assertEqual(expected_error_message, err_msg)
+    
+    # EM-14
+    def test_output_already_exist(self):
+        output_path = os.path.join(
+            self.general_output_dir, "already_existing_output.csv"
+        )
+        expected_error_message = "{} already exists. Please specify a new file path".format(output_path)
+
+        with open(output_path,"w") as outfile:
+            outfile.write("this output file already created")
+
+        with self.assertRaises(ValueError) as cm:
+            write_expression_output(
+                arg_output= output_path,
+                arg_delim=",",
+                arg_sql=False,
+                output_listdict_or_string=self.vizserver_data_mock_response["results"],
+                save_uncommon_delim_to_txt = False,
+                error_handler=self.common_value_error_handler
+            )
+
+        err_msg = str(cm.exception).strip()
+        self.assertEqual(expected_error_message, err_msg)
+
+    def test_output_is_directory(self):
+        output_path = os.path.join(
+            self.general_output_dir, "directory"
+        )
+        expected_error_message = "{} is a directory. Please specify a new file path".format(output_path)
+        os.mkdir(output_path)
+        with self.assertRaises(ValueError) as cm:
+            write_expression_output(
+                arg_output= output_path,
+                arg_delim=",",
+                arg_sql=False,
+                output_listdict_or_string=self.vizserver_data_mock_response["results"],
+                save_uncommon_delim_to_txt = False,
+                error_handler=self.common_value_error_handler
+            )
+
+        err_msg = str(cm.exception).strip()
+        self.assertEqual(expected_error_message, err_msg)
+
+    def test_incorrect_file_extension(self):
+        expected_error_message = 'File extension ".tsv" does not match delimiter ","'
```

Comment:
This test can be removed, because we don't (per spec) attempt to verify/update suffix if filename is already given.